iset.mm

```diff
@@ -135016,6 +135038,642 @@ the base set to the (extended) reals and which is nonnegative, symmetric
+       with underlying set ` X ` and distance function ` D ` .  (Contributed by
+       Mario Carneiro, 2-Sep-2015.) $)
+    isxms2 $p |- ( K e. *MetSp <->
+        ( D e. ( *Met ` X ) /\ J = ( MetOpen ` D ) ) ) $=
+      ( vx wcel cmopn cfv wceq wa cxmet ctopon cdm crn cuni adantl eleqtrd cxms
+      ctps isxms istps cv cbl ctg df-mopn dmmptss wrel toponmax simpl relelfvdm
+      mopnrel sylancr sseldi xmetunirn sylib eqid mopntopon syl toponuni eqtr4d
+      eqeltrd fveq2d ex eleq1 syl5ibr impbid syl5bb pm5.32ri bitri ) CUAICUBIZB
+      AJKZLZMADNKZIZVOMABCDEFGUCVOVMVQVMBDOKZIZVOVQDBCFEUDVOVSVQVOVSVQVOVSMZAAP
+      PZNKZVPVTANQRZIAWBIZVTJPZWCAHWCHUEUFKQUGKJHUHUIVTJUJDVNIAWEIUNVTDBVNVSDBI
+      VODBUKSVOVSULZTDAJUMUOUPAUQURZVTWADNVTWABRZDVTBWAOKZIWAWHLVTBVNWIWFVTWDVN
+      WIIWGAVNWAVNUSZUTVAVDWABVBVAVSDWHLVODBVBSVCVETVFVQVSVOVNVRIAVNDWJUTBVNVRV
+      GVHVIVJVKVL $.
+
+    $( Express the predicate " ` <. X , D >. ` is a metric space" with
+       underlying set ` X ` and distance function ` D ` .  (Contributed by NM,
+       27-Aug-2006.)  (Revised by Mario Carneiro, 24-Aug-2015.) $)
+    isms $p |- ( K e. MetSp <-> ( K e. *MetSp /\ D e. ( Met ` X ) ) ) $=
+      ( vf cv cds cfv cbs cxp cres cmet wcel cxms cms fveq2 syl6eqr wceq fveq2d
+      sqxpeqd reseq12d eleq12d df-ms elrab2 ) HIZJKZUHLKZUJMZNZUJOKZPADOKZPHCQR
+      UHCUAZULAUMUNUOULCJKZDDMZNAUOUIUPUKUQUHCJSUOUJDUOUJCLKDUHCLSFTZUCUDGTUOUJ
+      DOURUBUEHUFUG $.
+
+    $( Express the predicate " ` <. X , D >. ` is a metric space" with
+       underlying set ` X ` and distance function ` D ` .  (Contributed by NM,
+       27-Aug-2006.)  (Revised by Mario Carneiro, 24-Aug-2015.) $)
+    isms2 $p |- ( K e. MetSp <->
+        ( D e. ( Met ` X ) /\ J = ( MetOpen ` D ) ) ) $=
+      ( cxms wcel cmet cfv wa cxmet cmopn wceq cms isxms2 anbi1i isms metxmet
+      pm4.71ri an32 bitri 3bitr4i ) CHIZADJKIZLADMKIZBANKOZLZUFLZCPIUFUHLZUEUIU
+      FABCDEFGQRABCDEFGSUKUGUFLZUHLUJUFULUHUFUGADTUARUGUFUHUBUCUD $.
+
+    $( The topology component of a metric space coincides with the topology
+       generated by the metric component.  (Contributed by Mario Carneiro,
+       26-Aug-2015.) $)
+    xmstopn $p |- ( K e. *MetSp -> J = ( MetOpen ` D ) ) $=
+      ( cxms wcel ctps cmopn cfv wceq isxms simprbi ) CHICJIBAKLMABCDEFGNO $.
+
+    $( The topology component of a metric space coincides with the topology
+       generated by the metric component.  (Contributed by Mario Carneiro,
+       26-Aug-2015.) $)
+    mstopn $p |- ( K e. MetSp -> J = ( MetOpen ` D ) ) $=
+      ( cms wcel cmet cfv cmopn wceq isms2 simprbi ) CHIADJKIBALKMABCDEFGNO $.
+  $}
+
+  $( A metric space is a topological space.  (Contributed by Mario Carneiro,
+     26-Aug-2015.) $)
+  xmstps $p |- ( M e. *MetSp -> M e. TopSp ) $=
+    ( cxms wcel ctps ctopn cfv cds cbs cxp cres cmopn wceq eqid isxms simplbi )
+    ABCADCAEFZAGFAHFZQIJZKFLRPAQPMQMRMNO $.
+
+  $( A metric space is a topological space.  (Contributed by Mario Carneiro,
```

Comment:
A metric space is an extended metric space.

Reply:
Proposal adopted in set.mm and iset.mm.